### PR TITLE
hp80: HP82939 serial I/O module added

### PR DESCRIPTION
**Pull request by fulivi**
Hi,

I've just added the emulation of HP82939 serial I/O module for HP80 systems.
I needed to fix a small bug in INS8250 emulation because the module self-test was failing. Hopefully I didn't break anything in other systems with 8250.
ROM image for the module microcontroller will be emailed shortly to "code" address.
Thank you.
--F.Ulivi

### CODE REVIEW

File: src/devices/machine/1mb5.cpp [hunk 1]
```diff
@@ -76,18 +76,22 @@ READ8_MEMBER(hp_1mb5_device::cpu_r)
 	case 1:
 		// Read IB
 		res = m_ib;
-		m_ibf = false;
-		update_halt();
+		if (m_ibf) {
+			machine().scheduler().boost_interleave(attotime::from_usec(50) , attotime::from_usec(100));
+			space.device().execute().spin();
```
Comment (ajrhacker) [inline src/devices/machine/1mb5.cpp]: To me, this looks like an incredibly gross hack. There must be some better of way of handling the synchronization problem here.

Comment (fulivi) [reply src/devices/machine/1mb5.cpp]: Well, it's surely an incredibly gross hack but on the other hand I'm not a wizard on scheduler manipulation. What is your suggestion for a more elegant hack, please? The point here is that HP developers relied heavily on race conditions when they were sure of the outcome in order to save a few "wait" instructions. So I tried to force MAME scheduler into some tight interleaving to achieve the same result as real processors executing in parallel. When the scheduler uses the normal time quantum a processor gets too far ahead of the one it's interacting with.

Comment (MoochMcGee) [reply src/devices/machine/1mb5.cpp]: Maybe you could try `config.set_perfect_quantum()` in the machine config?

**Comment by fulivi** [comment]
Well, your suggestion about using "set_perfect_quantum" was the "perfect" one :)
Code is much cleaner now and scheduling apparently works. The only trouble I have now is that accelerating emulation through PageDown doesn't work anymore (speed goes to about 10 times the normal but emulation freezes).
I'm now wondering what would be the recommended use for "boost_interleave", it's clearly there for a reason..

**Pull request merged**